### PR TITLE
JBIDE-21422 - make server editor fields able to be edited for openshi…

### PR DESCRIPTION
**Pull request by robstryker**
…ft server

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java [hunk 1]
```diff
@@ -505,6 +509,17 @@ private void loadResources(Composite container, OpenShiftServerEditorModel model
 
 							@Override
 							protected IStatus run(IProgressMonitor monitor) {
+								model.setInitializing(true);
+								try {
+									String conUrl = server.getAttribute(OpenShiftServerUtils.ATTR_CONNECTIONURL, (String)null);
+									ConnectionURL conurl2 = ConnectionURL.forURL(conUrl);
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java]: is it null safe?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java [hunk 2]
```diff
@@ -505,6 +509,17 @@ private void loadResources(Composite container, OpenShiftServerEditorModel model
...
+									IConnection con = ConnectionsRegistrySingleton.getInstance().getByUrl(conurl2);
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java]: is it null safe?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java [hunk 3]
```diff
@@ -505,6 +509,17 @@ private void loadResources(Composite container, OpenShiftServerEditorModel model
...
+									model.setConnection(con);
+								} catch(UnsupportedEncodingException uee) {
+
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java]: multi catch please

**Comment by fbricon** [comment]
Can you add some tests for the new code pretty please?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServiceViewModel.java [hunk 4]
```diff
@@ -96,19 +97,19 @@ protected IService updateService(final IService service, final List<ObservableTr
 		return newService;
 	}
 
-	public void setConnections(List<Connection> connections) {
+	public void setConnections(List<IConnection> connections) {
 		update(this.connection, connections, this.service, this.serviceItems);
 	}
 
-	public Connection getConnection() {
+	public IConnection getConnection() {
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServiceViewModel.java]: ServiceViewModelTest doesn't compile anymore

**Comment by fbricon** [comment]
I have 2 connections to the cdk (admin/admin and openshift-dev/devel), but the connection combo doesn't select the openshift-dev one (blank instead)

**Comment by fbricon** [comment]
Also the source path input text is unusable. I recommend the browse buttons to be put to the next line, aligned right.

**Comment by robstryker** [comment]
Yes, I had mentioned in my standup that I need help with the connection not being pre-selected.   UI issues on source path will be looked into.

**Comment by adietish** [comment]
@robstryker ping me when you're on so that we can solve the connection selection issue.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java (hunk 1 above)
Comment (robstryker) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java]: no

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java (hunk 2 above)
Comment (robstryker) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java]: yes

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java (hunk 3 above)
Comment (robstryker) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java]: no.

**Comment by adietish** [comment]
we discussed the issued and fixed the remaining flaw with the selection in the connection combo. I tested it quickly and it worked for me. The PR though still needs me to test it more intesively. Will do tomorrow.

**Comment by fbricon** [comment]
It seems to work, except I can't set the route settings (seen in the advanced section of the wizard)

**Comment by robstryker** [comment]
> It seems to work, except I can't set the route settings (seen in the advanced section of the wizard)

I'd suggest this should be an enhancement and part of a separate jira. This jira is about making existing UI editable, and once we verify it works as expected we can isolate the route stuff separately.

**Comment by robstryker** [comment]
Opened https://issues.jboss.org/browse/JBIDE-22327   to target that new issue.

**Comment by fbricon** [comment]
Faire enough, so please rebase your PR so it can be applied.

**Comment by fbricon** [comment]
+1 once rebased

**Pull request merged**